simulation/chart.py

```diff
@@ -1,232 +1,427 @@
-        x_log (bool, optional): Use logarithmic scale to display x axis. Defaults to None.
-        y_label (str, optional): Label text to display next to y axis. Defaults to None.
-        y_range (Tuple[float,float], optional): Constant range(min,max value) of y axis. Defaults to None.
-        y_log (bool, optional): Use logarithmic scale for y axis. Defaults to None.
-    Returns:
-        PlotItem: a chart to contain plots
+class IndexedColor(NamedTuple):
+    """Color from a single index. Useful for stepping through a predefined list of colors.
+
+    Attributes:
+        index (int): predefined color index
+        hues (int): number of steps to generate
     """
-    chart = layout.addPlot(title=title)
-    if height is not None:
-        chart.setPreferredHeight(300)
-    legend = chart.addLegend(offset=None)
-    vb: pg.ViewBox = layout.addViewBox()
-    if legend_width is not None:
-        vb.setFixedWidth(legend_width)
-    legend.setParentItem(vb)
-    legend.anchor((0, 0), (0, 0))
-    chart.setClipToView(True)
 
-    if x_label is not None:
-        chart.setLabel("bottom", x_label)
-    if x_log is not None:
-        chart.setLogMode(x=x_log)
-    if x_range is not None:
-        chart.setXRange(*x_range)
+    index: int
+    hues: int
 
-    if y_label is not None:
-        chart.setLabel("left", y_label)
-    if y_log is not None:
-        chart.setLogMode(y=y_log)
-    if y_range is not None:
-        chart.setYRange(*y_range)
 
-    layout.nextRow()
+Color = Union[RGBAColor, NamedColor, IndexedColor, int, float, str]
+"""Color based on model, predefined values, single greyscale value (0.0 - 1.0) , color index or string representing color (“#RGB”,“#RGBA”,“#RRGGBB”,“#RRGGBBAA”)."""
 
-    return chart
 
+@singledispatch
+def _make_color(color: RGBAColor):  # pyright:ignore [reportGeneralTypeIssues]
+    """Convert RGBA color value to QColor"""
+    if color.A is None:
+        return pg.mkColor((color.R, color.G, color.B))
+    else:
+        return pg.mkColor((color.R, color.G, color.B, color.A))
 
-def add_line_plot(
-    chart: pg.PlotItem, name: str, color=None, width: Optional[float] = None, style=None
-) -> pg.PlotDataItem:
-    """Adds a line plot to the chart.
 
-    Args:
-        name (str): Name (legend title) of the plot
-        color (Any, optional): Line color of the plot. Defaults to None.
-        width (float, optional): Width of the plot line. Defaults to None.
-        style (Any, optional): Style of the plot line. Defaults to None.
-    """
-    config = {}
-    if color is not None:
-        config["color"] = color
+@singledispatch
+def _make_color(color: IndexedColor):  # pyright:ignore [reportGeneralTypeIssues]
+    """Convert Indexed color value to QColor"""
+    return pg.mkColor((color.index, color.hues))
+
+
+@singledispatch
+def _make_color(color: Union[int, float, str, None]):
+    if color is None:
+        return None
+    return pg.mkColor(color)
+
+
+class PenStyle(Enum):
```

Comment:
I don't typically use `Enum` in python. Are the attributes self-documenting? Else, document them here please!

Reply:
This is for better type support which is vague in pyqtgraph. I think the value definitions are well explaining themselves for a line type such as `DashDotLine`.

Reply:
But will the attributes be visible directly in the documentation, or will a user have to read your code?

Reply:
https://github.com/semiotic-ai/autoagora-agents/pull/21/commits/0f379f981088a9a3162a7ae1b81edb5798d4cc52

Reply:
Perfect! Thanks!